README.md

```diff
@@ -1,120 +1,240 @@
+
+If `transform_type` for a singleton is defined, it will no longer error as the value is already defined to be empty.
+
+### `Function`
+
+Attempting to hash a function errors by default. You can opt in to hashing a function
+by defining [`transform_type`](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.type_identifier), or by using [HashFunctions](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.HashFunctions). If `transform_type` is defined, functions will hash as if they were `UnorderedStructs`: functions can have fields if they are curried (e.g. `==(2)`), and so, for this reason, the fields are included in the hash by default.
+
+If a function value `fn` can be hashed in this way, so can types of the form `typeof(fn)`.
+
+### `Type`
+
+When a type is provided as a value (e.g. `stable_hash(Int; version=4)`) hashing it will error by default.
+
+You can opt in to hashing a given type as a value by defining [`transform_type_value`](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.transform_type_value) for your type, or by using [HashTypeValues](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.HashTypeValues)
+
+## Examples
+
+All of the following hash examples follow directly from the definitions above, but may not be so obvious to the reader.
+
+Most of the behaviors described below can be customized/changed by using your own hash [`@context`](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.@context), which can be passed as the second argument to [`stable_hash`](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.stable_hash).
+
+The order of NamedTuple pairs does not matter
+
+```julia
+stable_hash((;a=1,b=2); version=4) == stable_hash((;b=2,a=1); version=4)
+```
+
+Two structs with the same fields and name hash equivalently
+
+```julia
+module A
+    struct X
+        bar::Int
+        foo::Float64
+    end
+end
+
+module B
+    struct X
+        foo::Float64
+        bar::Int
+    end
+end
+
+stable_hash(B.X(2, 1); version=4) == stable_hash(A.X(1, 2); version=4)
+```
+
+Different array types with the same content hash to the same value.
+
+```julia
+stable_hash(view([1,2,3], 1:2); version=4) == stable_hash([1,2]; version=4)
+```
+
+Byte equivalent arrays of all `NumberType` values will hash to the same value.
+
+```julia
+stable_hash([0.0, 0.0]; version=4) == stable_hash([0, 0]; version=4)
```

Comment:
We should add this example:
```julia
stable_hash([0.0f0, 0.0f0]; version=4) != stable_hash([0, 0]; version=4)
```

Also, even though the bytes are the same, since the size is hashed, we have:
```julia
stable_hash([0.0f0, 0.0f0]; version=4) != stable_hash([0]; version=4)
```